R/transformCounts.R

```diff
@@ -317,6 +325,25 @@ setMethod("transformAssay", signature = c(x = "SummarizedExperiment"),
     return(mat)
 }
 
+####################################.apply_css###################################
+# This function applies cumulative sum scaling (CSS) to the abundance table.
+.calc_css <- function(mat, ...){
+  # If mat is a DelayedArray, convert it to a regular matrix for computation
+  if (inherits(mat, "DelayedArray")) {
+    mat <- as.matrix(mat)
+  }
+  
+  # Calculate the cumulative sum for each sample
+  cum_sum <- apply(mat, 2, cumsum)
+
+  # Find the scaling factor for each sample, typically the 75th percentile of the cumulative sum
+  scaling_factors <- apply(cum_sum, 2, function(x) quantile(x, 0.75))
```

Comment:
The 0.75 value should be something that the user can change if they wish. Put this to the function arguments, and 0.75 can be the default when it is not specified. See other functions for similar stuff on how it is best implemented.

Reply:
Perhaps pass it with `...` as it affect only this transformation